metricsaggregator/src/main/java/io/dockstore/metricsaggregator/helper/RunExecutionAggregator.java

```diff
@@ -0,0 +1,97 @@
+
+    /**
+     * Aggregates TaskExecutions that belong to a single workflow run into a workflow-level RunExecution
+     * @param taskExecutionsForOneWorkflowRun
+     * @return
+     */
+    Optional<RunExecution> getWorkflowExecutionFromTaskExecutions(TaskExecutions taskExecutionsForOneWorkflowRun);
+
+    /**
+     * Aggregates workflow executions into an aggregated metric.
+     * @param workflowExecutions
+     * @return
+     */
+    Optional<M> getAggregatedMetricFromWorkflowExecutions(List<RunExecution> workflowExecutions);
+
+    /**
+     * Aggregates a list of aggregated metrics into one aggregated metric.
+     * @param aggregatedMetrics
+     * @return
+     */
+    Optional<M> getAggregatedMetricsFromAggregatedMetrics(List<M> aggregatedMetrics);
+
+    /**
+     * Returns a list of RunExecutions where the execution metric is not null.
+     * @param runExecutions
+     * @return
+     */
+    default List<E> getNonNullMetricsFromRunExecutions(List<RunExecution> runExecutions) {
+        return runExecutions.stream()
+                .map(execution -> getMetricFromRunExecution(execution))
+                .filter(Objects::nonNull)
+                .toList();
+    }
+
+    /**
+     * Aggregate metrics from all submissions in the ExecutionsRequestBody.
+     * This method uses the runExecutions, taskExecutions, and aggregatedExecutions from ExecutionRequestBody to create an aggregated metric.
+     * Metrics are aggregated by:
+     * <ol>
+     *     <li>Aggregating task executions, provided via ExecutionRequestBody.taskExecutions, into workflow executions.</li>
+     *     <li>Aggregating workflow executions,submitted via ExecutionRequestBody.runExecutions and workflow executions that were aggregated from task executions, into an aggregated metric.
+     *     <li>Aggregating the list of aggregated metrics, submitted via ExecutionRequestBody.aggregatedExecutions and the aggregated metric that was aggregated from workflow executions, into one aggregated metric.</li>
+     * </ol>
+     * @param allSubmissions
+     * @return
+     */
+    default Optional<M> getAggregatedMetricFromAllSubmissions(ExecutionsRequestBody allSubmissions) {
+        final List<RunExecution> workflowExecutions = new ArrayList<>(allSubmissions.getRunExecutions());
+
+        // If task executions are present, calculate the workflow RunExecution containing the overall workflow-level execution time for each list of tasks
+        if (!allSubmissions.getTaskExecutions().isEmpty()) {
+            final List<RunExecution> calculatedWorkflowExecutionsFromTasks = allSubmissions.getTaskExecutions().stream()
+                    .map(taskExecutions -> getWorkflowExecutionFromTaskExecutions(taskExecutions))
+                    .filter(Optional::isPresent)
+                    .map(Optional::get)
+                    .toList();
+            workflowExecutions.addAll(calculatedWorkflowExecutionsFromTasks);
+        }
```

Comment:
This is the change in the main aggregation function that aggregates task executions into workflow executions. The rest of the code in this function is existing logic that was moved over from each individual `getAggregated<Metric>` function in `AggregationHelper`